File: backend/src/main/java/com/contentplatform/backend/infrastructure/jpa/entity/SitemapUrlCheckEntity.java
package com.contentplatform.backend.infrastructure.jpa.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "sitemap_url_checks")
public class SitemapUrlCheckEntity {
    @Id
    @Column(name = "id", nullable = false, length = 36)
    private String id;

    @Column(name = "tenant_id", nullable = false, length = 36)
    private String tenantId;

    @Column(name = "url", nullable = false, length = 2048)
    private String url;

    @Column(name = "last_checked_at", nullable = false)
    private Instant lastCheckedAt;

    @Column(name = "http_status")
    private Integer httpStatus;

    @Column(name = "error_message", columnDefinition = "text")
    private String errorMessage;

    protected SitemapUrlCheckEntity() {
    }

    public SitemapUrlCheckEntity(String id,
                                 String tenantId,
                                 String url,
                                 Instant lastCheckedAt,
                                 Integer httpStatus,
                                 String errorMessage) {
        this.id = id;
        this.tenantId = tenantId;
        this.url = url;
        this.lastCheckedAt = lastCheckedAt;
        this.httpStatus = httpStatus;
        this.errorMessage = errorMessage;
    }
}

File: backend-nestjs/src/entities/sitemap-url-check.entity.ts
import { Column, Entity, Index, PrimaryGeneratedColumn } from 'typeorm';

@Entity({ name: 'sitemap_url_checks' })
@Index(['tenantId', 'url'])
export class SitemapUrlCheckEntity {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column({ name: 'tenant_id', type: 'varchar', length: 36 })
  tenantId!: string;

  @Column({ type: 'varchar', length: 2048 })
  url!: string;

  @Column({ name: 'last_checked_at', type: 'timestamptz' })
  lastCheckedAt!: Date;

  @Column({ name: 'http_status', type: 'integer', nullable: true })
  httpStatus!: number | null;

  @Column({ name: 'error_message', type: 'text', nullable: true })
  errorMessage!: string | null;
}

